### PR TITLE
ORV2-458 CRUD Company Profile part2

### PR DESCRIPTION
**Pull request by praju-aot**
# Description

Please provide a summary of the change and the issue fixed. Please include relevant context. List dependency changes.

Second iteration of Company Profile CRUD

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] npm run test
- [x] npm run test:e2e
- [x] Run docker compose run --build -d, verify container starts successfully
- [x] Test the swagger Apis at http://:5000/api

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have already been accepted and merged

## Further comments

---

Thanks for the PR!

Any successful deployments (not always required) will be available below.
[Backend - Vehicles](https://onroutebc-186-backend-vehicles.apps.silver.devops.gov.bc.ca/) available
[Frontend](https://onroutebc-186-frontend.apps.silver.devops.gov.bc.ca/) available

Once merged, code will be promoted and handed off to following workflow run.
[Main Merge Workflow](https://github.com/bcgov/onroutebc/actions/workflows/merge-main.yml)

### CODE REVIEW

File: backend/vehicles/src/modules/company/company-profiles/profiles/company-profile.profile.ts [hunk 1]
```diff
@@ -49,7 +49,7 @@ export class CompanyProfile extends AutomapperProfile {
             }
           }),
         ),
-        forMember((d) => d.companyDirectory, fromValue('bbceid')),
+        forMember((d) => d.companyDirectory, fromValue('BBCEID')),
```
Comment (john-fletcher-aot) [inline backend/vehicles/src/modules/company/company-profiles/profiles/company-profile.profile.ts:52]: Wonder if we should use the standard directory names from WebADE? BCEID being business BCeID, MYID being basic BCeID? If we change this it'll have to be changed throughout.

Comment (praju-aot) [reply backend/vehicles/src/modules/company/company-profiles/profiles/company-profile.profile.ts:52]: Hi John,

I have sent an email to Chris Robinson asking the same. I will create a new PR upon his confirmation. Also, these values will be grabbed from token. I have hardcoded them now as we haven't integrated login with our application

Comment (john-fletcher-aot) [reply backend/vehicles/src/modules/company/company-profiles/profiles/company-profile.profile.ts:52]: Sounds good, we can figure out the specifics of the IDs a bit later on.

**Review by john-fletcher-aot — APPROVED** [review]
Reviewed and approved

**Pull request merged**